File: kotlin/solid_principles/solid/Subscriber.kt
package solid

class Subscriber : SubscriptionNotificationService {
    private var email : String?= null

    override fun subscribeToNewProductsAvailability() {
        println("Subscribing to new products availability...")
    }

    override fun subscribeToSMSNotification() {
        println("Subscribing to SMS notification...")
    }
}

File: typescript/solid-principles/Subscriber.ts
import { SubscriptionNotificationService } from "./subscription-notification-service";

export class Subscriber implements SubscriptionNotificationService{

    private email: string;
    
    subscribeToNewProductsAvailability(): void {
        console.log("Subscribing to new products availability...");
    }

    subscribeToSMSNotifications(): void {
        console.log("Subscribing to SMS notifications...");
    }

    
}